src/_zkapauthorizer/_storage_server.py

```diff
@@ -292,12 +307,25 @@ def remote_slot_testv_and_readv_and_writev(
         r_vector,
     ):
         """
-        Pass-through after a pass check to ensure clients can only allocate
-        storage for mutable shares if they present valid passes.
+        Perform a test-and-set on a number of shares in a given slot.
+
+        Optionally, also read some data to be returned before writing any
+        changes.
+
+        If storage-time will be allocated by the operation then validate the
+        given passes and ensure they are of sufficient quantity to pay for the
+        allocation.
+
+        Specifically, passes are required in the following cases:
 
-        :note: This method can be used both to allocate storage and to rewrite
-            data in already-allocated storage.  These cases may not be the
-            same from the perspective of pass validation.
+        * If shares are created then a lease is added to them.
+          Passes are required for the full size of the share.
+
+        * If shares without unexpired leases are written then a lease is added to them.
+          Passes are required for the full size of the shares after the write.
+
+        * If shares with unexpired leases are made larger.
+          Passes are required for the difference in price between the old and new size.
```

Comment:
As noted in the PR comment, a client will lose some value here (inversely proportional to how much of a remaining lease period there is on those shares).

Reply:
Added a note to this docstring in 9abf3a50